src/box/space.c

```diff
@@ -852,6 +852,9 @@ generic_space_swap_index(struct space *old_space, struct space *new_space,
 {
 	SWAP(old_space->index_map[old_index_id],
 	     new_space->index_map[new_index_id]);
+	/* Swap read gaps back - they belong to the old space. */
+	rlist_swap(&old_space->index_map[old_index_id]->read_gaps,
```

Comment:
Elaborated in commit message:
```
Note that we swap indexes on DDL before the old space is invalidated, so
all index read gaps are actually moved to the new space. It's actually
incorrect but didn't break anything - we delete all transaction's
trackers when it is prepared or aborted, and we used to abort all
in-progress transactions on DDL. But with the new approach, we need
those gaps in the old space in order to find all space readers, so the
commit swaps read gaps back when moving space indexes.
```

Reply:
Done:
```c
/*
 * Swap read gaps back - they belong to the old space. It is important
 * since the old space is going to be invalidated and transactional
 * metadata is required to do it correctly.
 */
```